x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,114 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
+)
+
+// Simple model, send tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/v3/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DistributeToProviderValidatorSet(ctx sdk.Context) error {
+
+	ltbh, err := k.GetLastTransmissionBlockHeight(ctx)
+	if err != nil {
+		return err
+	}
+	bpdt := k.GetBlocksPerDistributionTransmission(ctx)
+	curHeight := ctx.BlockHeight()
+
+	if (curHeight - ltbh.Height) < bpdt {
+		// not enough blocks have passed for  a transmission to occur
+		return nil
+	}
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	fpTokens := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+
+	// split the fee pool, send the consumer's fraction to the consumer redistribution address
+	fracStr := k.GetConsumerRedistributeFrac(ctx)
+	frac, err := sdk.NewDecFromStr(fracStr)
+	if err != nil {
+		return err
+	}
+	decFPTokens := sdk.NewDecCoinsFromCoins(fpTokens...)
+	consRedistrTokens, _ := decFPTokens.MulDec(frac).TruncateDecimal()
```

Comment:
Why are we doing this logic here? It seems simpler to me that the child chain application decides how much of the fee goes into their community pool, and how much goes into the ccv account. And then the full ccv account gets sent to the provider.

It also reduces the code inside ccv

Reply:
I've replied to your comment in this thread (let's continue discussion there). https://github.com/cosmos/interchain-security/pull/32#discussion_r859101977